File: proton-sdk/incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/telemetry/DownloadError.kt
package me.proton.drive.sdk.telemetry

enum class DownloadError {
    UNRECOGNIZED,
    SERVER_ERROR,
    NETWORK_ERROR,
    DECRYPTION_ERROR,
    INTEGRITY_ERROR,
    RATE_LIMITED,
    VALIDATION_ERROR,
    HTTP_CLIENT_SIDE_ERROR,
    UNKNOWN,
}

File: proton-sdk/client/cs/src/Proton.Drive.Sdk/Telemetry/DownloadError.cs
namespace Proton.Drive.Sdk.Telemetry;

// Numeric values must match the DownloadError enum in proton.drive.sdk.proto.
public enum DownloadError
{
    ServerError = 0,
    NetworkError = 1,
    DecryptionError = 2,
    IntegrityError = 3,
    RateLimited = 4,
    HttpClientSideError = 5,
    Unknown = 6,
    ValidationError = 7,
}
